packages/react-components/src/components/chart/baseChart.tsx

```diff
@@ -117,14 +118,18 @@ const BaseChart = ({ viewport, queries, size = { width: 500, height: 500 }, ...o
   // determine the set option settings
   const settings = useChartSetOptionSettings(dataStreams);
 
-  //handle dataZoom updates, which are dependent on user events and viewportInMS changes
+  // handle dataZoom updates, which are dependent on user events and viewportInMS changes
   useDataZoom(chartRef, viewportInMs);
 
+  // handle chart event updates
+  const eventOptions = useHandleChartEvents(chartRef);
+
   // set all the options on the echarts instance
   useEChartOptions(
     chartRef,
     {
       ...convertedOptions,
+      ...eventOptions,
```

Comment:
convertedOptions also has options for tooltip, you probably want to seperate out tooltip options or merge them

Reply:
The convertedOptions tooltip does not specify the `show` parameter. Even if it did we'd want the eventOptions to override it, so this should be safe to do.

Reply:
I see `eventOptions` is totally overwriting the `convertedOptions.tooltip`. Will merge the two tooltips.